src/CMakeLists.txt

```diff
@@ -18,6 +18,12 @@ list(APPEND SRCS args.cc baseclient.cc client.cc http.cc request.cc response.cc
 add_library(miniocpp STATIC ${SRCS})
 target_link_libraries(miniocpp ${requiredlibs})
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"
```

Comment:
The actual fix is
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73eee94..5389961 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,13 +40,16 @@ IF (${CMAKE_BINARY_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
   MESSAGE(FATAL_ERROR "Please create a separate build directory and call CMake again")
 ENDIF (${CMAKE_BINARY_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
 
-IF(CMAKE_COMPILER_IS_GNUCC )
-  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Werror")
-ENDIF()
-
 # Look for required libraries
 SET(requiredlibs)
 
+IF(CMAKE_COMPILER_IS_GNUCC)
+  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Werror")
+  IF(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0 AND NOT MINGW)
+    list(APPEND requiredlibs stdc++fs)
+  ENDIF()
+ENDIF()
+
 find_package(CURL REQUIRED)
 IF(CURL_FOUND)
   INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
```

Reply:
> The actual fix is
> 
> ```diff
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 73eee94..5389961 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -40,13 +40,16 @@ IF (${CMAKE_BINARY_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
>    MESSAGE(FATAL_ERROR "Please create a separate build directory and call CMake again")
>  ENDIF (${CMAKE_BINARY_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
>  
> -IF(CMAKE_COMPILER_IS_GNUCC )
> -  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Werror")
> -ENDIF()
> -
>  # Look for required libraries
>  SET(requiredlibs)
>  
> +IF(CMAKE_COMPILER_IS_GNUCC)
> +  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Werror")
> +  IF(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0 AND NOT MINGW)
> +    list(APPEND requiredlibs stdc++fs)
> +  ENDIF()
> +ENDIF()
> +
>  find_package(CURL REQUIRED)
>  IF(CURL_FOUND)
>    INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
> ```

I had modfied as this.

Reply:
The modification is wrong. There is no need to have two separate `IF(CMAKE_COMPILER_IS_GNUCC)`